src/mixer/basetrackplayer.cpp

```diff
@@ -134,8 +136,37 @@ BaseTrackPlayerImpl::BaseTrackPlayerImpl(QObject* pParent,
             SLOT(slotWaveformZoomSetDefault(double)));
 
     m_pPreGain = std::make_unique<ControlProxy>(group, "pregain", this);
-    // BPM of the current song
 
+    m_pShiftTimesEarlier = std::make_unique<ControlPushButton>(
+            ConfigKey(group, "shift_times_earlier"));
+    connect(m_pShiftTimesEarlier.get(),
+            &ControlObject::valueChanged,
+            this,
+            [=](double value) { slotShiftTimesButton(value, -1 * kShiftTimesOffsetMillis); });
```

Comment:
with [=] all sort of bad things can happen. Please catch explicit what you need. In this case [this].
kShiftTimesOffsetMillis from the anonymous namespace should accessible without capture. 
Also below.